File: app/src/main/java/me/pwcong/jpstart/manager/SharedPreferenceManager.java
package me.pwcong.jpstart.manager;

import android.content.Context;
import android.content.SharedPreferences;
import android.support.annotation.Nullable;

import me.pwcong.jpstart.App;
import me.pwcong.jpstart.conf.Constants;

/**
 * Created by Pwcong on 2016/9/27.
 */

public class SharedPreferenceManager {

    private static SharedPreferenceManager instance = null;

    private final SharedPreferences sharedPreferences;

    private SharedPreferenceManager() {

        sharedPreferences = App.getInstance().getSharedPreferences(Constants.PREF_NAME, Context.MODE_PRIVATE);

    }

    public static synchronized SharedPreferenceManager getInstance() {

        if (instance == null) {
            instance = new SharedPreferenceManager();
        }

        return instance;
    }

    public String getString(String key, @Nullable String defValue) {

        return sharedPreferences.getString(key, defValue);

    }

    public void putString(String key, String value) {

        sharedPreferences.edit().putString(key, value).apply();

    }

    public int getInt(String key, int defValue) {

        return sharedPreferences.getInt(key, defValue);

    }

    public void putInt(String key, int value) {
        sharedPreferences.edit().putInt(key, value).apply();

    }

    public boolean getBoolean(String key, boolean defValue) {
        return sharedPreferences.getBoolean(key, defValue);
    }

    public void putBoolean(String key, boolean value) {
        sharedPreferences.edit().putBoolean(key, value).apply();
    }

}

File: app/src/main/java/com/github/pwcong/jpstart/manager/SharedPreferenceManager.kt
package com.github.pwcong.jpstart.manager

import android.content.Context
import android.content.SharedPreferences

import com.github.pwcong.jpstart.App
import com.github.pwcong.jpstart.constant.Constant

class SharedPreferenceManager private constructor() {
    private val sharedPreferences: SharedPreferences = App.getInstance()
        .getSharedPreferences(Constant.PREF_NAME, Context.MODE_PRIVATE)

    fun getString(key: String?, defValue: String?): String? {
        return sharedPreferences.getString(key, defValue)
    }

    fun putString(key: String?, value: String?) {
        sharedPreferences.edit().putString(key, value).apply()
    }

    fun getInt(key: String?, defValue: Int): Int {
        return sharedPreferences.getInt(key, defValue)
    }

    fun putInt(key: String?, value: Int) {
        sharedPreferences.edit().putInt(key, value).apply()
    }

    fun getBoolean(key: String?, defValue: Boolean): Boolean {
        return sharedPreferences.getBoolean(key, defValue)
    }

    fun putBoolean(key: String?, value: Boolean) {
        sharedPreferences.edit().putBoolean(key, value).apply()
    }

    companion object {
        private var instance: SharedPreferenceManager? = null

        @Synchronized
        fun getInstance(): SharedPreferenceManager {
            if (instance == null) {
                instance = SharedPreferenceManager()
            }

            return instance!!
        }
    }
}